### DISCUSSION TITLE
character corruption during message transport

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5744] [Q&A] [answered]
There is a patient in our EHR (Epic) and her name contains a letter with an accent over it (n with a tilde above it).  When an ADT message comes out of Epic and goes into mirth (connection via TCP listener) that letter gets converted to a null character and appears as a square instead of an 'n'. 

If I copy the message in Epic and paste it into the Mirth channel, it works fine with no corruption. Is there some setting that would affect how character codes pass via TCP into Mirth?

[Mirth Character Corruption.docx](https://github.com/nextgenhealthcare/connect/files/11149163/Mirth.Character.Corruption.docx)

**Comment by jonbartels** [comment]
Related discussion - https://github.com/nextgenhealthcare/connect/discussions/5457 . No solution but does this detail describe the same problem you are having?

**Comment by rutledj** [comment]
Yep. Looks like the same issue except mine is TCP vs file.

**Reply by jonbartels** [reply]
What are your source listener character encoding options set to?

**Reply by rutledj** [reply]
It is set to default. Changed to UTF-8 but that didn't fix it.

**Reply by jonbartels** [reply]
Is the message getting mangled in transmission or only on display in MC? 

I'm not sure if I can explain myself very well. If you query the database in the d_mcXX table for your message content, is the content correctly encoded and displayed? 

If you have Mirth receive the message and send it back out to a dumb tcp app like `netcat` (`nc -l 9090`) do the diacritic marks send OK?

Finally - can you get a packet capture of the data sent to Mirth and ensure that it is transmitted with the correct hex values?

I recall some issue where MC would not display the messages properly but would transmit them properly.

**Reply by rutledj** [reply]
It is corrupted when it goes to the final destination.  I don't have network access do a packet capture.

**Comment by rutledj** [comment]
We were able to configure the Epic interface to send out UTF-8.  Is that the Mirth default for TCP listeners (when the encoding drop-down is set to 'Default')?

**Reply by ab-20-m** [reply]
The Default charset varies based on the OS for Java 17 and earlier. [JEP 400 UTF-8 by Default](https://openjdk.org/jeps/400) changed the behavior in Java 18+.

From the JEP:

**Description**
In JDK 17 and earlier, the default charset is determined when the Java runtime starts. On macOS, it is UTF-8 except in the POSIX C locale. On other operating systems, it depends upon the user's locale and the default encoding, e.g., on Windows, it is a codepage-based charset such as windows-1252 or windows-31j. The method java.nio.charsets.Charset.defaultCharset() returns the default charset. A quick way to see the default charset of the current JDK is with the following command:

**Comment by rutledj** [comment, marked as answer]
FYI, I was able to correct this issue by setting the charset in Epic to UTF-8. By default it is Windows 1252
